swap/swap.go

```diff
@@ -344,6 +368,42 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque")
+	}
+
+	if !cheque.Equal(p.getPendingCheque()) {
+		p.logger.Warn("ignoring confirm msg, wrong cheque", "got", cheque, "expected", p.getPendingCheque())
+		return
+	}
+
+	if err := p.setLastSentCheque(cheque); err != nil {
+		p.Drop("persistence error")
```

Comment:
Why would we drop the peer if our node cannot save the cheque? I would propose to shit down Swarm with this i/o error.

Reply:
It's not entirely clear to me how to shut down swarm and I didn't want to ignore the error entirely, so I used drop because that's what the other protocols do on io error as well.

Reply:
Requesting @janos to give his opinion here. 
After an IO error, is it sufficient to just drop the peer (as done by @ralph-pichler here) or should Swarm be stopped entirely? If we need to stop Swarm, how to do this safely?

Reply:
we don't support a shutdown invoked by some high level components. you could `panic` and risk data corruption with the databases since a shutdown is not invoked when panicking. dropping the peer in this case is the safer option because that way, if you're getting local IO errors and you start dropping peers - you isolate the problem to your local node and technically you do not participate in the network anymore (ideally). I'd stick with dropping peers at the moment since node behavior with IO errors is something we have not tested yet and it is quite an involved task in its own

Reply:
A very interesting question. I agree with @acud, as we do not have a way to shut down the node, except with panic. If this error happens, we will drop the peer, but it is very likely that the same I/O error will happen in all next calls, so we will drop all peers, practically removing our node from the network. And maybe recovering if the I/O problem gets resolved. This could happen if the disk is full, for example. Maybe we could drop all peers if this happens which would prevent other peers to step into the same problem. I am not sure how easy is to do so currently. If it requires non trivial changes, this drop is ok for now.

Reply:
see #1515